lib/nx_gzlib.c

```diff
@@ -0,0 +1,295 @@
+/* nx_gzlib.c -- implement gz* functions
+ *
+ * NX-GZIP compression accelerator user library
+ * implementing zlib compression library interfaces
+ *
+ * Copyright (C) IBM Corporation, 2022
+ *
+ * Licenses for GPLv2 and Apache v2.0:
+ *
+ * GPLv2:
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ *
+ * Apache v2.0:
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <errno.h>
+#include "nx_zlib.h"
+
+#define DEF_MEM_LEVEL 8
+#define MAX_LEN 4096
+
+gzFile nx_gzopen_(const char* path, int fd, const char *mode);
+
+typedef struct gz_state {
+	gzFile gz;
+	int fd;
+	int err;
+	z_stream strm;
+	bool deflate;
+} *gz_statep;
+
+int nx_gzclose(gzFile file)
+{
+	gz_statep state = (gz_statep) file;
+	z_streamp stream = &(state->strm);
+	unsigned char* next_out = malloc(sizeof(char)*MAX_LEN);
+	int ret;
+
+	if(file == NULL){
+		errno = EINVAL;
+		return Z_STREAM_ERROR;
+	}
+
+	if (state->deflate) {
+		/* Finish the stream.  */
+		stream->next_in = Z_NULL;
+		stream->avail_in = 0;
+		do {
+			stream->next_out = next_out;
+			stream->avail_out = MAX_LEN;
+
+			nx_deflate(stream, Z_FINISH);
+			write(state->fd, next_out, MAX_LEN - stream->avail_out);
+		} while (stream->avail_out == 0);
+
+		ret = nx_deflateEnd(stream);
+	} else
+		ret = nx_inflateEnd(stream);
+
+	free(next_out);
+	free(state);
+
+	return ret;
+}
+
+gzFile nx_gzdopen(int fd, const char *mode)
+{
+	return nx_gzopen_(NULL, fd, mode);
+}
+
+gzFile nx_gzopen(const char* path, const char *mode)
+{
+	return nx_gzopen_(path, 0, mode);
+}
+
+gzFile nx_gzopen_(const char* path, int fd, const char *mode)
+{
+	gz_statep state;
+	gzFile file;
+	int err, strategy = Z_DEFAULT_STRATEGY, level = Z_DEFAULT_COMPRESSION;
+	char* digit;
+	void* stream;
+
+	state = malloc(sizeof(struct gz_state));
+	memset(state, 0, sizeof(struct gz_state));
```

Comment:
This line is being added in the 4th commit. Doesn't it belong with the first one that adds `nx_gzopen`?